contexts/network-settings.tsx

```diff
@@ -0,0 +1,434 @@
+
+import { isSameSet } from "helpers/array";
+import { isColorsSimilar } from "helpers/colors";
+import { DefaultNetworkSettings } from "helpers/custom-network";
+
+import { Color, Icon, NetworkSettings, Theme } from "interfaces/network";
+
+import useApi from "x-hooks/use-api";
+import useNetworkTheme from "x-hooks/use-network";
+import useOctokitGraph from "x-hooks/use-octokit-graph";
+
+const { publicRuntimeConfig } = getConfig();
+
+const NetworkSettingsContext = createContext<NetworkSettings | undefined>(undefined);
+
+const ALLOWED_PATHS = ["/new-network", "/[network]/account/my-network/settings"];
+
+const LIMITS = {
+  percentageNeededForDispute: { min: +publicRuntimeConfig?.networkConfig?.disputesPercentage },
+  draftTime: {
+    min: +publicRuntimeConfig?.networkConfig?.reedemTime?.min,
+    max: +publicRuntimeConfig?.networkConfig?.reedemTime?.max,
+  },
+  disputableTime: {
+    min: +publicRuntimeConfig?.networkConfig?.disputableTime?.min,
+    max: +publicRuntimeConfig?.networkConfig?.disputableTime?.max,
+  },
+  councilAmount: {
+    min: +publicRuntimeConfig?.networkConfig?.councilAmount?.min,
+    max: +publicRuntimeConfig?.networkConfig?.councilAmount?.max
+  }
+};
+
+export const NetworkSettingsProvider = ({ children }) => {
+  const router = useRouter();
+  
+  const [github, setGithub] = useState(DefaultNetworkSettings.github);
+  const [tokens, setTokens] = useState(DefaultNetworkSettings.tokens);
+  const [details, setDetails] = useState(DefaultNetworkSettings.details);
+  const [treasury, setTreasury] = useState(DefaultNetworkSettings.treasury);
+  const [parameters, setParameters] = useState(DefaultNetworkSettings.parameters);
+  const [tokensLocked, setTokensLocked] = useState(DefaultNetworkSettings.tokensLocked);
+  const [isSettingsValidated, setIsSettingsValidated] = useState(DefaultNetworkSettings.isSettingsValidated);
+  
+  const { activeNetwork } = useNetwork();
+  const { service: DAOService } = useDAO();
+  const { DefaultTheme } = useNetworkTheme();
+  const { wallet, user } = useAuthentication();
+  const { getUserRepositories } = useOctokitGraph();
+  const { getNetwork, searchRepositories, repositoryHasIssues } = useApi();
+
+  const isCreating = useMemo(() => router.pathname === "/new-network", [router.pathname]);
+  const needsToLoad = useMemo(() => ALLOWED_PATHS.includes(router.pathname), [router.pathname]);
+
+  const Fields = {
+    amount: {
+      setter: (value: number) => setTokensLocked(previous => ({ ...previous, amount: value })),
+      validator: (locked: number, needed: number) => needed > 0 && locked >= needed
+    },
+    name: {
+      setter: (value: string) => setDetails(previous => ({
+        ...previous, 
+        validated: false, 
+        name: { 
+          value, 
+          validated: undefined 
+        } 
+      })),
+      validator: async (value: string): Promise<boolean | undefined> => {
+        let validated = undefined;
+  
+        if (value.trim() !== "")
```

Comment:
I've noticed we're allowing _spaces_ on the network name (There's a network literally called `"network name"` -- we need to replace the spaced into dashes, after trimming:

```js
value = value.trim().replace(/\s+/g,'-');
```
```

Reply:
@moshmage do we need to replace just because of the URL or really the name can't have spaces? if it's just because of the URL, I can make the replacement only at the time of assembling the url

Reply:
@vhcsilva both -- if we allow spaces but replace the dash, there could be something like "new network" _and_ "new-network" that would be different but be shown the same way and then would create a conflict when we try to query it